Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml

```diff
@@ -348,6 +348,7 @@
     <Include ref='Geometry/VeryForwardData/data/ppstrackerMaterials/v1/ppstrackerMaterials.xml'/>
     <Include ref='Geometry/VeryForwardData/data/CTPPS_Pixel_Module/v1/CTPPS_Pixel_Module.xml'/>
     <Include ref='Geometry/VeryForwardData/data/CTPPS_Pixel_Module_2x2/v1/CTPPS_Pixel_Module_2x2.xml'/>
+    <Include ref="Geometry/VeryForwardData/data/dd4hep/SpecPar_for_CTPPS_Pixel_Module_2x2.xml"/>
```

Comment:
I expect this change to fail geometry unit tests. The Run3 geometry is now generated from dictionaries, so the update should be made in: https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/dict2021Geometry.py.

Reply:
Hmm an issue I see is that this is only a DD4hep file, hence it is only added to DD4hep XML include file.

Reply:
Basically these are specpars which only make sense to add for DD4hep.

Reply:
If they are added to dict2021, they would also be added for old DD, which we do no want.

Reply:
@ianna do you have any suggestions for how to handle this? We haven't encountered this case before, as far as I'm aware.

Reply:
I'd think it would not harm to have the file in both DD and DD4hep scenarios. The first will just ignore it.

Reply:
Hmm in practice old DD complains if there are paths like:
`world_volume/OCMS_1/CMSE_1/RP_147_Right_Station_1/RP_147_Right_Station_Vacuum_5_1/RP_147_Right_Station_Vacuum_3_Far_1/RP_box_primary_vacuum_10103/RP_box_secondary_vacuum_0/Envelop_1/RPixWafer_1`
in SpecPar section, even if these paths are only relevant for DD4hep.
I found a trick by replacing it by 
`//RP_147_Right_Station_1/RP_147_Right_Station_Vacuum_5_1/RP_147_Right_Station_Vacuum_3_Far_1/RP_box_primary_vacuum_10103/RP_box_secondary_vacuum_0/Envelop_1/RPixWafer_1`
And in that case oldDD properly ignores it.

So I can add that to 
`Geometry/VeryForwardData/data/CTPPS_Pixel_Module_2x2/v1/CTPPS_Pixel_Module_2x2.xml`
A priori I thought it would be cleaner to not touch oldDD XMLs at all, but ok this works out.

Reply:
Hmm actually 
`//RP_147_Right_Station_1/RP_147_Right_Station_Vacuum_5_1/RP_147_Right_Station_Vacuum_3_Far_1/RP_box_primary_vacuum_10103/RP_box_secondary_vacuum_0/Envelop_1/RPixWafer_1`
does not work either for old DD.

Hmm if we want the exact same geo dictionary, hence the exact same XMLs, hence SpecPar paths, for oldDD and DD4hep, this is gonna be tricky.
Because the volumes names in the navigation for old DD and DD4hep are slightly different.

Anyway this SpecPar section has always been a complete pain, I just went for a solution which does not use SpecPar at all.

Reply:
I finished addressing all review points, will push the commits once I have finished re-doing all the tests.